src/figuresClasses.ts

```diff
@@ -1,11 +1,85 @@
-export interface Figure {}
+type Color = 'red' | 'green' | 'blue';
+type Shape = 'triangle' | 'circle' | 'rectangle';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  color: Color;
+  shape: Shape;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error(
+        'The side of the figure cannot be less than or equal to zero',
+      );
+    }
+
+    const maxLength: number = Math.max(a, b, c);
+    const sumLength: number = a + b + c;
+
+    if (maxLength >= sumLength - maxLength) {
+      throw new Error("Sides 1, 2, and 3 can't form a triangle");
```

Comment:
The error message could be more informative by specifying the sides that cannot form a triangle. This would help in debugging and understanding the issue better.